mesh_segmenter/src/mesh_segmenter.cpp

```diff
@@ -69,22 +73,44 @@ void MeshSegmenter::segmentMesh()
 
   included_indices_.clear();
 
-  for(int i = 0; i < size; ++i)
+  // Loop over all of the cells
+  for (int i = 0; i < size; ++i)
   {
     // if current index is not used, perform segmentation with it and insert the results into the list
-    if(used_cells->IsId(i) == -1)
+    if (used_cells->IsId(i) == -1)
     {
       vtkSmartPointer<vtkIdList> linked_cells = vtkSmartPointer<vtkIdList>::New();
       linked_cells = segmentMesh(i);
 
       // save indices found
-      included_indices_.push_back(linked_cells);
-      for(int j = 0; j < linked_cells->GetNumberOfIds(); ++j)
+      if (linked_cells->GetNumberOfIds() > min_cluster_size_)
       {
-        used_cells->InsertNextId(linked_cells->GetId(j));
+        included_indices_.push_back(linked_cells);
+
+        for (int j = 0; j < linked_cells->GetNumberOfIds(); ++j)
+        {
+          used_cells->InsertNextId(linked_cells->GetId(j));
+        }
       }
     }
   }
+
+  // Anything still not used in a segment is an "edge" - There has to be an easier way...
```

Comment:
One could convert the mesh to a half edge mesh and walk the half edge meshes to find the boarders
Also, there is an example in the vtk examples pages
https://lorensen.github.io/VTKExamples/site/Cxx/Meshes/BoundaryEdges/

Reply:
Well in this case edge is in quotes. It actually would not have to include the half-edge edge. It just includes anything not in one of the other segments. This will be cells where is segment they belong to is too small to satisfy the minimum segment size constraint